core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/puppeteer_utils.js

```diff
@@ -0,0 +1,240 @@
+// Copyright 2023 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Utility File for the Acceptance Tests.
+ */
+
+const puppeteer = require('puppeteer');
+const testConstants = require('./testConstants.js');
+const { showMessage } = require('./showMessageUtils.js');
+
+const rolesEditorTab = testConstants.URLs.RolesEditorTab;
+const roleEditorInputField = 'input.e2e-test-username-for-role-editor';
+const roleEditorButtonSelector = 'e2e-test-role-edit-button';
+const rolesSelectDropdown = 'mat-select-trigger';
+
+module.exports = class puppeteerUtilities {
+  page;
+  browserObject;
+  userHasAcceptedCookies = false;
+
+  /**
+   * This is a function that opens a new browser instance for the user.
+   * @returns {Promise<puppeteer.Page>} - Returns a promise that resolves
+   * to a Page object controlled by Puppeteer.
+   */
+  async openBrowser() {
+    /* Currently, headless is set to false and the page viewport
+       is maximized so that it would be easy for the developers
+       to debug easily while testing.
+       We can remove these settings before merging as we have
+       to run the tests in headless mode. */
```

Comment:
Should we allow customizing this through some test runner script? Like you either do `start_acceptance_tests` or `start_acceptance_tests --headless`

Reply:
yes.. this can be added in the runner scripts for acceptance tests. can you please help in implementing this?